specification/dartLangSpec.tex

```diff
@@ -8969,24 +8969,28 @@ \subsection{Symbols}
 a valid declaration name or a valid library name in a Dart program.
 
 \begin{grammar}
-<symbolLiteral> ::= `#' (<operator> | (<identifier> (`.' <identifier>)*))
+<symbolLiteral> ::=
+  `#' (<operator> | <identifier> (`.' <identifier>)* | \VOID{})
 \end{grammar}
 
 \LMHash{}%
 A symbol literal \code{\#\id} where \id{} is an identifier
-that does not begin with an underscore (`\code{\_}'),
-evaluates to an instance of \code{Symbol} representing the identifier \id.
-All occurrences of \code{\#\id} evaluate to the same instance
+that does not begin with an underscore (`\code{\_}') or \VOID{}
```

Comment:
Maybe rewrite as "is \VOID{} or an identifier that does not ..." to avoid the ambiguity of whether "or \VOID{}" binds to the "is" or the "not begin with". Also potentially ambiguous whether we're saying that `void` is an identifier.

Maybe it can be done with commas too, or by adding "either"/"or".

"where \id{} is either an identifier that does not begin with an underscore (`\code{\_}'), or is the reserved word \VOID{},"

Comma after \VOID{} at least.

Reply:
The wording was actually rather sloppy. I revised the section as a whole, such that we only specify canonicalization twice (once for the private identifier, once for all other cases). I also moved the static analysis (just one sentence) from the end to the beginning, to bring about the ordering of topics that we've used in all other sections that were substantially revised.